### ISSUE TITLE
I think an additional method of filtering is needed

### ISSUE THREAD

**Issue by waleadebayo** [benjjneb/decontam#26]
Hi thanks for the good work.
I think there is need to add another filter apart from frequency and prevalence to decontam (am also trying to write some scripts for this) but I should perhaps assume I am wrong  first.

I believe both freq and previous or combined with Fisher or any of the combinations do not remove an OTU or ASV that is only found in one "Control Sample" (blanks).

Like I was saying on q2, I have attached shots here
![screen shot 2018-08-24 at 18 39 10](https://user-images.githubusercontent.com/19531978/44909794-9aa31700-ad18-11e8-84f3-401851e8e8a9.png)

![screen shot 2018-08-24 at 18 36 24](https://user-images.githubusercontent.com/19531978/44909795-9bd44400-ad18-11e8-86cc-f6c1fe402d57.png)

**Comment by benjjneb** [comment]
You are correct. This is a consequence of a larger decision that the decontam package makes to not attempt classification on any variants present in just one sample. Those variants are assigned an NA score and given the default classification (which is non-contaminant for `isContaminant`).

Fundamentally this is because decontam uses patterns across samples to identify contaminants, and is definitely the right choice for the frequency method. However, for the prevalence method, it might makes sense to call something found in just one negative control sample as a contaminant...

Right now I'm thinking that is a better 1-sample behavior for the prevalence method...

**Comment by waleadebayo** [comment]
I did realise this (the search for patterns) process  through which decontam does its job after checking the result and reading the paper, which explains the title of my posting. I think it could be added to decontam  to cover such contaminant. piece of cake, I think???

**Comment by benjjneb** [comment]
Yes, I think this could be added with no negative consequences. Tagging this as enhancement.

**Comment by pstev** [comment]
Would this mean that any OTU only present in one non-control sample would also be given the default classification? Or does it only apply to control samples?

**Comment by benjjneb** [comment]
> Would this mean that any OTU only present in one non-control sample would also be given the default classification? Or does it only apply to control samples?

The idea would that OTUs present only in control samples would be flagged as contaminants, even if they appear just once. No change to current behavior would results for OTUs that appear even once in a non-control sample (i.e. an OTU present in one non-control sample would still receive the default classification of non-contaminant).

**Comment by pstev** [comment]
So in my case I'm using isNotContaminant, as I've got really low biomass samples - would an OTU only present in one non-control sample be classified as a contaminant using this method, or would it be identified as a non-contaminant as it is present in a higher prevalence in samples than controls?

**Comment by waleadebayo** [comment]
@pstev 
First, I think you need to understand that we were discussing is another way (that decontam is not programmed to do now) that might also help to complement what it can do now (as you might expect developers are always looking for improvement). So, it is not exactly related to whether you are using isContaminant or isNotContaminant function.   

Basically, one of the methods put in there looks specifically at groups of blanks/negative controls (samples you expect nothing should be there if everything was pure and perfectly clean), so the current discussion was on creating another method that also looks at individuals too.

I will advise you to go on use the tool as it is. of course, developers can later decide if it is indeed beneficial to add what we are pointing out here and probably another version.

**Comment by pstev** [comment]
Ok, thanks both!
